python/api/config/serializers.py

```diff
@@ -0,0 +1,111 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=redefined-builtin
+
+from rest_framework import serializers
+from rest_framework.fields import JSONField
+from rest_framework.reverse import reverse
+
+import logrotate
+from cm.adcm_config import ui_config, restore_cluster_config
+from cm.api import update_obj_config
+from cm.errors import AdcmEx, AdcmApiEx
+
+
+class ConfigURL(serializers.HyperlinkedIdentityField):
+    def get_url(self, obj, view_name, request, format):
+        kwargs = {
+            'object_type': obj.prototype.type,
+            f'{obj.prototype.type}_id': obj.id
+        }
+        return reverse(view_name, kwargs=kwargs, request=request, format=format)
+
+
+class ConfigVersionURL(serializers.HyperlinkedIdentityField):
+    def get_url(self, obj, view_name, request, format):
+        kwargs = {
+            'object_type': obj.object_type,
+            f'{obj.object_type}_id': obj.object_id,
+            'version': obj.id
+        }
+        return reverse(view_name, kwargs=kwargs, request=request, format=format)
+
+
+class HistoryCurrentPreviousConfigSerializer(serializers.Serializer):
+    history = serializers.SerializerMethodField()
+    current = serializers.SerializerMethodField()
+    previous = serializers.SerializerMethodField()
+
+    def get_history(self, obj):
+        view_name = 'config-history'
+        return ConfigURL(read_only=True, view_name=view_name).get_url(
+            obj, view_name, self.context['request'], format=None)
+
```

Comment:
You don't need to do it so complicated and use SerializerMethodField. Just write:

history = ConfigURL(read_only=True, view_name=view_name)

Reply:
Fixed in [https://github.com/arenadata/adcm/pull/563/commits/2ff1cbcf6ee5573650cd2fdf0928c4ac22360223](url)